### PR TITLE
Small edits to the circuit code

### PR DESCRIPTION
**Pull request by cathieyun**
cleanups:
- doc updates
- fixme notes
- update benchmark, doctest code
- rename funcs
etc etc

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -112,21 +112,19 @@ impl VecPoly3 {
...
-    // MUST call l_poly.inner_product(r_poly).
-    pub fn inner_product(&self, rhs: &VecPoly3) -> Poly6 {
-        let l = self;
-        let r = rhs;
-
+    /// Compute an inner product of `lhs`, `rhs` which have the property that:
+    /// - `lhs.0` is zero;
+    /// - `rhs.2` is zero;
+    /// This is the case in the constraint system proof.
+    pub fn special_inner_product(lhs: &Self, rhs: &Self) -> Poly6 {
```
Comment (oleganza) [inline src/util.rs]: 👍 for reflecting this in the fn name.

File: benches/r1cs.rs [hunk 2]
```diff
@@ -69,128 +69,111 @@ For K = 1:
...
+/// The inputs and outputs are all tuples of the `Variable, Assignment`, where the `Assignment`
+/// can be either assigned as `Value::Scalar` or unassigned as `Missing`.
+pub fn fill_cs<CS: ConstraintSystem>(
+    cs: &mut CS,
+    x: Vec<(Variable, Assignment)>,
+    y: Vec<(Variable, Assignment)>,
+) -> Result<(), KShuffleError> {
+    let one = Scalar::one();
+    let z = cs.challenge_scalar(b"k-scalar shuffle challenge");
+    let neg_z = -z;
```
Comment (oleganza) [inline benches/r1cs.rs]: nitpick: we can make the code a bit cleaner by just passing `z` and subtracting it when needed - the difference in performance of `Add` and `Sub` is really insignificant if there is any.

File: src/lib.rs [hunk 3]
```diff
@@ -29,7 +29,7 @@ mod util;
 
 #[doc(include = "../docs/notes.md")]
 mod notes {}
-mod circuit_proof;
+mod constraint_system;
```
Comment (oleganza) [inline src/lib.rs]: (Maybe in the separate PR or even later) My bikeshed preference for names is: 

1. Rename private module `constraint_system` into `cs`.
2. Rename public module with reexports from `r1cs` into `cs`. All crate-private identifiers could be made `pub(crate)` or we can do `use cs as cs_internal` inside `mod.rs` so we avoid using `pub(crate)` and can create another module `cs` for public re-exports.
3. Rename `R1CSProof` into `ConstraintSystemProof`.

Comment (oleganza) [reply src/lib.rs]: I'm also down with `s/cs/constraint_system/` if you prefer the spelled out name. But for both private and public modules.

Comment (oleganza) [reply src/lib.rs]: How about internal “cs” and external “constraint_system”?

**Comment by cathieyun** [comment]
addressed suggestions, will merge after the CI tests pass, unless there is more feedback!

**Comment by hdevalence** [comment]
LGTM

File: src/lib.rs (hunk 3 above)
Comment (hdevalence) [reply src/lib.rs]: I think that "R1CS" is a good name and that we should keep it.

**Pull request merged**